vdr/src/contracts/anoncreds/revocation_registry.rs

```diff
@@ -0,0 +1,845 @@
+///
+/// # Params
+/// - `client`: [LedgerClient] - client connected to the network where contract will be executed
+/// - `id`: [RevocationRegistryId] - id of Revocation Registry Definition to resolve
+/// - `to_timestamp`: [u64] - timestamp of the status list to be retrieved
+///
+/// # Returns
+///   rev_reg_delta: [Option<RevocationRegistryDelta>] - Resolved RevocationRegistryDelta object
+pub async fn fetch_revocation_delta(
+    client: &LedgerClient,
+    id: &RevocationRegistryDefinitionId,
+    to_timestamp: u64,
+) -> VdrResult<Option<RevocationRegistryDelta>> {
+    let rev_reg_entries = fetch_revocation_registry_entries(&client, &id, to_timestamp).await?;
+
+    if rev_reg_entries.is_empty() {
+        return Ok(None);
+    }
+
+    //TODO: sets? vectors?
+    let mut issued: HashSet<u32> = HashSet::new();
+    let mut revoked: HashSet<u32> = HashSet::new();
+
+    let accum = rev_reg_entries
+        .last()
+        .unwrap()
+        .rev_reg_entry
+        .current_accumulator
+        .clone();
+
+    for rev_reg_entry in rev_reg_entries.into_iter() {
+        for issue in rev_reg_entry.rev_reg_entry.issued {
+            issued.insert(issue);
+
+            revoked.remove(&issue);
+        }
+
+        for revocation in rev_reg_entry.rev_reg_entry.revoked {
+            issued.remove(&revocation);
+
+            revoked.insert(revocation);
+        }
+    }
+
+    let mut issued: Vec<u32> = issued.into_iter().collect();
+    issued.sort();
+    let mut revoked: Vec<u32> = revoked.into_iter().collect();
+    revoked.sort();
+
+    Ok(Some(RevocationRegistryDelta {
+        revoked,
+        issued,
+        accum,
+    }))
+}
+
+/// Function to build a new revocation delta to save on the blockchain from a previous revocation delta
+///
+/// # Params
+/// - `revocation_registry_status_list`: [Vec<RevocationState>] - new desired Revocation Status
+/// List
+/// - `previous_delta`: [Option<RevocationRegistryDelta>] - previous delta saved on the blockchain
+/// - `accumulator`: [String] - new accumulator
+/// - `timestamp`: [String] - timestamp
+///
+/// # Returns
+///   rev_reg_delta: [RevocationRegistryEntryData] - RevocationRegistryDelta object
+fn build_latest_revocation_registry_entry_data(
+    revocation_registry_status_list: &Vec<RevocationState>,
+    previous_delta: &Option<RevocationRegistryDelta>,
+    accumulator: String,
+    timestamp: u64,
+) -> VdrResult<RevocationRegistryEntryData> {
+    let mut issued: Vec<u32> = Vec::new();
+    let mut revoked: Vec<u32> = Vec::new();
+
+    match previous_delta {
+        Some(previous_delta) => {
+            previous_delta.validate(revocation_registry_status_list.len() as u32 - 1)?;
+            // Check whether the revocationStatusList entry is not included in the previous delta issued indices
+            for (index, entry) in (0u32..).zip(revocation_registry_status_list.iter()) {
```

Comment:
It seems, can be replaced with: `revocation_registry_status_list.iter().enumerate()`